array.c

```diff
@@ -3554,55 +3674,99 @@ static VALUE rb_ary_bsearch_index(VALUE ary);
 
 /*
  *  call-seq:
- *     ary.bsearch {|x| block }  -> elem
- *
- *  By using binary search, finds a value from this array which meets
- *  the given condition in O(log n) where n is the size of the array.
```

Comment:
I think we should keep this performance characteristic note (`O(log n)`) in the updated documentation, maybe at the end?

Reply:
Restored.